plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -175,7 +175,7 @@ Object sanitizeArrayAndRecordMutation(@NonNull Object[] objects, @CheckForNull E
             this.isUnmodifiedBySanitization = false;
             return NotStoredReason.OVERSIZE_VALUE;
         }
-        List inputList = List.of(objects);
+        List<Object> inputList = Arrays.asList(objects);
```

Comment:
Also found this from a usages search that did not look obviously safe.

Reply:
[`@NonNull Object[] objects](https://github.com/jenkinsci/workflow-cps-plugin/pull/642/files#diff-72a51c4c67a1c8eaed43038451cf5b39f7221a5c486a7e8464b72481af91d0c2R173)`  would imply otherwise. either that annotation is wrong or the caller is violating this.

Reply:
https://github.com/jenkinsci/workflow-cps-plugin/blob/7c908a0a2d9b16b68a0a6f010f183cc8c34b145e/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java#L173 refers to the array, which is not null. But `List.of` fails also when an _element_ is null (which the annotation does not mention).

In practice I doubt there could be null elements, because arrays would be of `Describable` subtypes which would require real values, but I would rather not find out. Safely handling inputs with null elements would require additional work.